### PR TITLE
graph-store: speed up +validate-graph by using +turn and no longer virtualizing

### PR DESCRIPTION
**Pull request by tacryt-socryp**
In my test, the time to validate a graph went from 56 seconds to 1 second when joining UC General.

Perf Before:

Screen Shot 2021-03-31 at 11 45 12 AM

Perf After:
Screen Shot 2021-03-31 at 11 28 09 AM

### CODE REVIEW

**Review by joemfb — CHANGES_REQUESTED** [review]
Nice graphs!

As we discussed elsewhere, `+roll` is still preferable to `+turn`, as it avoids allocating the unneeded output list (the inverse of urbit/landscape#674).

It'd be better if this were factored to not depend on unused evaluations, and to avoid the redundant assertions at the call sites. I think the simplest way to do that would be to assert `?>  ?=(& (roll ...))` inside `+validate-graph`, and then produce the graph.

File: pkg/arvo/app/graph-store.hoon [hunk 1]
```diff
@@ -699,13 +700,12 @@
...
-        =(%& -:(mule |.((vale:dais [atom post.node]))))
+    ^-  ?
+    ?&  ?=(^ (vale:dais [atom post.node]))
         ?-  -.children.node
-            %empty  %.y
-            %graph  ^$(graph p.children.node)
-        ==
-    ==
+          %empty  %.y
+          %graph  ^$(graph p.children.node)
```
[the remark this answers is not in the document]
Comment (tacryt-socryp) [reply pkg/arvo/app/graph-store.hoon:707]: ahh that's a great point! Thanks.

**Review by joemfb — APPROVED** [review]
LGTM

**Review by philipcmonk — APPROVED** [review]

**Comment by matildepark** [comment]
@tacryt-socryp This still needs to retarget pre-release.

**Comment by ohAitch** [comment]
Crashmolds were a mistake UwU

**Pull request merged**